ignition_ros2_control/src/ignition_ros2_control_plugin.cpp

```diff
@@ -0,0 +1,468 @@
+  this->dataPtr->stop_ = false;
+  auto spin = [this]()
+    {
+      while (rclcpp::ok() && !this->dataPtr->stop_) {
+        this->dataPtr->executor_->spin_once();
+      }
+    };
+  this->dataPtr->thread_executor_spin_ = std::thread(spin);
+
+  RCLCPP_DEBUG_STREAM(
+    this->dataPtr->node->get_logger(), "[Ignition ROS 2 Control] Setting up controller for [" <<
+      model.Name(_ecm) << "] (Entity=" << _entity << ")].");
+
+  // Get list of enabled joints
+  auto enabledJoints = this->dataPtr->GetEnabledJoints(
+    _entity,
+    _ecm);
+
+  if (enabledJoints.size() == 0) {
+    RCLCPP_DEBUG_STREAM(
+      this->dataPtr->node->get_logger(), "[Ignition ROS 2 Control] There are no available Joints.");
+    return;
+  }
+
+  // Read urdf from ros parameter server then
+  // setup actuators and mechanism control node.
+  // This call will block if ROS is not properly initialized.
+  std::string urdf_string;
+  std::vector<hardware_interface::HardwareInfo> control_hardware;
+  try {
+    urdf_string = this->dataPtr->getURDF();
+    control_hardware = hardware_interface::parse_control_resources_from_urdf(urdf_string);
+  } catch (const std::runtime_error & ex) {
+    RCLCPP_ERROR_STREAM(
+      this->dataPtr->node->get_logger(),
+      "Error parsing URDF in ignition_ros2_control plugin, plugin not active : " << ex.what());
+    return;
+  }
+
+  std::unique_ptr<hardware_interface::ResourceManager> resource_manager_ =
+    std::make_unique<hardware_interface::ResourceManager>();
+
+  try {
+    this->dataPtr->robot_hw_sim_loader_.reset(
+      new pluginlib::ClassLoader<ignition_ros2_control::IgnitionSystemInterface>(
+        "ignition_ros2_control",
+        "ignition_ros2_control::IgnitionSystemInterface"));
+  } catch (pluginlib::LibraryLoadException & ex) {
+    RCLCPP_ERROR(
+      this->dataPtr->node->get_logger(), "Failed to create robot simulation interface loader: %s ",
+      ex.what());
+    return;
+  }
+
+  for (unsigned int i = 0; i < control_hardware.size(); ++i) {
+    std::string robot_hw_sim_type_str_ = control_hardware[i].hardware_class_type;
+    auto gazeboSystem = std::unique_ptr<ignition_ros2_control::IgnitionSystemInterface>(
+      this->dataPtr->robot_hw_sim_loader_->createUnmanagedInstance(robot_hw_sim_type_str_));
+
+    if (!gazeboSystem->initSim(
+        this->dataPtr->node,
+        enabledJoints,
+        control_hardware[i],
+        _ecm))
+    {
+      RCLCPP_FATAL(
+        this->dataPtr->node->get_logger(), "Could not initialize robot simulation interface");
+      return;
+    }
+
+    resource_manager_->import_component(std::move(gazeboSystem));
+  }
+  // Create the controller manager
+  RCLCPP_INFO(this->dataPtr->node->get_logger(), "Loading controller_manager");
+  this->dataPtr->controller_manager_.reset(
+    new controller_manager::ControllerManager(
+      std::move(resource_manager_),
+      this->dataPtr->executor_,
+      "controller_manager"));
```

Comment:
This name should be configurable through URDF where `ignition_ros2_control` plugin is loaded.